### PR TITLE
Gene expression

### PR DESCRIPTION
**Pull request by kellrott**
Adding initial code for RNA matrix parser. Tested against GTEX. May need some additional options for formatting aliquot gids based on values found in the matrix.

### CODE REVIEW

File: transform/rna-matrix/transform-rna-matrix.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+from bmeg.edge import ExpressionOf
+from bmeg.emitter import JSONEmitter
+
+
+def matrix_parse(handle, emitter, args):
+    df = csv.reader(handle, delimiter="\t")
+    samples = next(df)[1:]
+    genes = []
+    # write values out to temp matrix
+    t = open("tmp_matrix.data", "wb")
```
Comment (buchanae) [inline transform/rna-matrix/transform-rna-matrix.py]: Would an in-memory pandas transpose work? These expressions matrices usually fit in memory.

File: Dockerfile [hunk 2]
```diff
@@ -11,6 +11,9 @@ COPY transform/gdc/*.py /opt/
 # Ensembl Gene / Transcript / Exon
 COPY transform/ensembl/*.py /opt/
 
+# Copy RNA Transform code
+COPY transform/rna-matrix/transform-rna-matrix.py /opt/
```
Comment (buchanae) [inline Dockerfile]: I'm concerned that the repo layout is changing inside the docker container. The layout of the tree should stay the same.

File: transform/rna-matrix/transform-rna-matrix.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--matrix', type=str, help='')
+    parser.add_argument("--output-prefix", default="rna")
+    parser.add_argument("--scale", default="TPM")
+    parser.add_argument("--method", default="Illumina HiSeq")
+    parser.add_argument("--source", default="NA")
+    parser.add_argument("--gene-label-split", default=None, nargs=2)
+    parser.add_argument("--gz", action="store_true", default=False)
```
Comment (buchanae) [inline transform/rna-matrix/transform-rna-matrix.py]: We've decided to make transform scripts hard-coded to their inputs. Most transformers are specific to one input file, and its easier for us to collaborate when we don't need to learn and understand a long list of CLI arguments.

A reusable expression matrix parser should live in `src/bmeg/expression.py`.

File: transform/rna-matrix/transform-rna-matrix.py (hunk 1 above)
Comment (buchanae) [reply transform/rna-matrix/transform-rna-matrix.py]: I tried pandas and it used 6GB of memory but provided suuuuper slow access to the columns. Probably not the right solution.

**Comment by buchanae** [comment]
I've made a few improvements:

- made GCT code reusable
- added transformer for CCLE GCT 
- add ExpressionMetric, which captures TPM, RPKM, etc. into an enum
- bugfixes for GCT stuff

**Comment by buchanae** [comment]
Added TCGA transcript-level TPM code. Untested

**Comment by kellrott** [comment]
```
  File "/Users/ellrott/workspaces/bmeg-etl-2/venv/lib/python3.7/site-packages/bmeg-0.1.0-py3.7.egg/bmeg/utils.py", line 55, in wrapper
TypeError: __init__() got an unexpected keyword argument 'scale'
```

**Pull request merged**